api/tests/unit/interfaces/controllers/security-controller_test.js

```diff
@@ -569,4 +570,110 @@ describe('Unit | Interfaces | Controllers | SecurityController', () => {
     });
   });
 
+  describe('#checkUserBelongsToOrganizationOrHasRolePixMaster', () => {
+
+    let belongsToOrganizationStub;
+    let hasRolePixMasterStub;
+
+    beforeEach(() => {
+      belongsToOrganizationStub = sinon.stub(checkUserBelongsToOrganizationUseCase, 'execute');
+      hasRolePixMasterStub = sinon.stub(checkUserHasRolePixMasterUseCase, 'execute');
+    });
+
+    context('Successful case', () => {
+      const request = {
+        auth: {
+          credentials: {
+            accessToken: 'valid.access.token',
+            userId: 1234
+          }
+        },
+        params: { id: 5678 }
+      };
+
+      it('should authorize access to resource when the user is authenticated and belongs to organization', async () => {
+        // given
+        belongsToOrganizationStub.resolves(true);
+        hasRolePixMasterStub.resolves(false);
+
+        // when
+        const response = await securityController.checkUserBelongsToOrganizationOrHasRolePixMaster(request, hFake);
+
+        // then
+        expect(response.source).to.equal(true);
+      });
+
+      it('should authorize access to resource when the user is authenticated and is PIX_MASTER', async () => {
+        // given
+        belongsToOrganizationStub.resolves(false);
+        hasRolePixMasterStub.resolves(true);
+
+        // when
+        const response = await securityController.checkUserBelongsToOrganizationOrHasRolePixMaster(request, hFake);
+
+        // then
+        expect(response.source).to.equal(true);
+      });
+
+      it('should authorize access to resource when the user is authenticated and belongs to organization and is PIX_MASTER', async () => {
+        // given
+        belongsToOrganizationStub.resolves(true);
+        hasRolePixMasterStub.resolves(true);
+
+        // when
+        const response = await securityController.checkUserBelongsToOrganizationOrHasRolePixMaster(request, hFake);
+
+        // then
+        expect(response.source).to.equal(true);
+      });
+    });
+
+    context('Error cases', () => {
+      const request = {
+        auth: {
+          credentials: {
+            accessToken: 'valid.access.token',
+            userId: 1234
+          }
+        },
+      };
+
+      it('should forbid resource access when user was not previously authenticated', async () => {
+        // given
+        delete request.auth.credentials;
+
+        // when
+        const response = await securityController.checkUserBelongsToOrganizationOrHasRolePixMaster(request, hFake);
+
+        // then
+        expect(response.statusCode).to.equal(403);
+        expect(response.isTakeOver).to.be.true;
+      });
+
+      it('should forbid resource access when user does not belong to organization or has role PIXMASTER', async () => {
```

Comment:
c'est pas "should forbid resource access when user does not belong to organization and does not have role PIXMASTER"